src/Analysis/pipeline/parameter_types.py

```diff
@@ -0,0 +1,323 @@
+    def __init__(self, float=None):
+        self.float = float
+        """value"""
+    
+    def to_string(self):
+        """Presentation of type in json yaml"""
+        return float 
+      
+    def is_set(self):
+        """
+        return if structure contain real data
+        """
+        return  self.float is not None
+
+    def assigne(self,  value):
+        """
+        Assigne appropriate python variable to data     
+        """
+        if isinstance(value, BaseData):
+            self.float = float(value.value) 
+        else:
+            self.float = float(value)
+
+    def getter(self):
+        """
+        Return appropriate python variable to data     
+        """
+        return self.float
+
+class CompositeData(metaclass=abc.ABCMeta):
+    """
+    Abbstract class od port types, that define user for
+    validation and translation to string
+    """
+    def __init__(self, subtype, *args):
+        self.subtype = subtype
+        """Contained type"""
+        self.list = []
+        for arg in args:
+            self.list.append(arg)
+   
+    @abc.abstractmethod 
+    def merge(self, composite):
+        """merge to composite types of same type"""
+        pass
+        
+    @abc.abstractmethod 
+    def reduce_values(self, *values):
+        """delete all values diferent from set"""
+        pass
+
+    def add(self, *values):
+        """add values to composite"""
+        for arg in values:
+            self.list.append(arg)
+    
+    def to_string(self, value):
+        """Presentation of type in json yaml"""
+        if self.subtype is None:
+            return None
+        try:
+            res = "[\n"
+            first = True
+            for subvalue in value:
+                if first:
+                    first = False
+                else:
+                    res += ",\n"
+                res += self.subtype.to_string(subvalue)
+            res += "]\n"
+        except:
+            return None    
+            
+    def contain(self, data):
+        """
+        return if structure contain set data
+        """
+        return True
+        
+    def is_set(self):
+        """
+        return if structure contain real data
+        """
+        return True
+ 
+class Struct(CompositeData):
+    """
+    Object
+    """
+    def __init__(self, **kwargs):
+        """Contained type"""
+        for name, value in kwargs.items():
+            setattr(self, name, value)
+   
+    def merge(self, composite):
+        """merge to composite types of same type"""
+        for name, value in composite.__dict__.items():
+            setattr(self, name, value)
+        
+    def reduce_values(self, *values):
+        """delete all values diferent from set"""
+        dell_list=[]
+        for name, value in self.__dict__.items():
+            if value not in values: 
+                dell_list.append(name)
+        for name in dell_list:
+            delattr(self, name)
+
+    def reduce_keys(self, *values):
```

Comment:
Tohle ted nechci moc řešit, v budoucnu použiji co budu moc, přejmenuji, zbytek vymažu.